### PR TITLE
#23603 : Allow users to write code when creating a file

### PR DESCRIPTION
**Pull request by jcastro-dotcms**
### Proposed Changes
* Allow users to write code when creating a file. A new method was added to the Temp File API and the Temp File Endpoint to allow back-end users to "upsert" a temporary file.
* The Text Area field is displayed when a new File Asset is created. This way, Users can either upload a binary file, or start typing its contents in the Text Area field.
* The initial mechanism used to create the content of the File Asset is the same one used when we're editing an existing text File Asset.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileAPI.java [hunk 1]
```diff
@@ -252,6 +245,34 @@ public DotTempFile createTempFileFromUrl(final String incomingFileName,
 
   }
 
+  public DotTempFile updateTempFile(final String tempFileId, final String incomingFileName,
+                                    final HttpServletRequest request, final InputStream inputStream) throws DotSecurityException {
+    if ("new".equalsIgnoreCase(tempFileId)) {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileAPI.java:250]: Use a constant

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileAPI.java [hunk 2]
```diff
@@ -252,6 +245,34 @@ public DotTempFile createTempFileFromUrl(final String incomingFileName,
 
   }
 
+  public DotTempFile updateTempFile(final String tempFileId, final String incomingFileName,
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileAPI.java:248]: Doc?

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileResource.java [hunk 3]
```diff
@@ -185,6 +182,119 @@ private void saveMultipleBinary(final FormDataMultiPart body, final HttpServletR
...
+    @JSONP
+    @NoCache
+    @Produces("application/octet-stream")
+    @Consumes(MediaType.APPLICATION_JSON)
+    public final Response updateTempResource(@Context final HttpServletRequest request,
+                                             @Context final HttpServletResponse response,
+                                             @PathParam("tempFileId") final String tempFileId,
+                                             final PlainTextFileForm body) {
+        this.verifyTempResourceEnabled();
+        new WebResource.InitBuilder(request, response).requiredAnonAccess(AnonymousAccess.WRITE).rejectWhenNoUser(true).init();
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileResource.java:204]: I am not sure if this service must be open to anonymous, the normal upload is ok, but this one is for very specific case, again not sure if it should or not @wezell ?

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileResource.java [hunk 4]
```diff
@@ -185,6 +182,119 @@ private void saveMultipleBinary(final FormDataMultiPart body, final HttpServletR
...
+
+    private void savePlainTextFile(final String tempFileId, final PlainTextFileForm body, final HttpServletRequest request,
+                                    final OutputStream outputStream, final ObjectMapper objectMapper) {
+        final CompletionService<Object> completionService = this.createCompletionService(2, 5, 100000);
+        final HttpServletRequest statelessRequest = RequestUtil.INSTANCE.createStatelessRequest(request);
+        // this triggers the save
+        final InputStream in = new ByteArrayInputStream(body.fileContent().getBytes());
+        final Future<Object> future = this.updateTempFileFuture(completionService, statelessRequest, 1,
+                body.fileName(), in, tempFileId);
+        /*final Future<Object> future = completionService.submit(() -> {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileResource.java:247]: remove this

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileResource.java (hunk 3 above)
Comment (jcastro-dotcms) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileResource.java]: Done!

**Comment by jcastro-dotcms** [comment]
Closing this PR as it will not be included in `release-23.03`

**Pull request closed without merging**